src/createApp.ts

```diff
@@ -334,7 +334,10 @@ function resolveFactory(type: FactoryTypes, { factory, instance }: ItemDefinitio
 	}
 
 	if (typeof instance === 'object') {
-		return Promise.resolve(() => instance);
+		// FIXME: Factories can return instances without having to return a promise for them, why does that suddenly
+		// cause TypeScript to throw a fit?!
+		const promise = Promise.resolve(instance);
+		return Promise.resolve(() => promise);
```

Comment:
Why are you returning a function that returns an instance, why are you just not resolving to the instance?

```typescript
return Promise.resolve(instance);
```

In the second, you are "tricking" it by returning a promise that returns a promise that resolves to an instance TypeScript it typed to unwrap promises, but no unwrap functions.